spring-pulsar-test/src/main/java/org/springframework/pulsar/test/support/Conditions.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.test.support;
+
+import java.util.stream.Stream;
+
+import org.apache.pulsar.client.api.Message;
+
+import org.springframework.util.Assert;
+
+/**
+ * Exposes a set of commonly used conditions to be used in {@link PulsarTestConsumer}.
+ *
+ * @author Jonas Geiregat
+ */
+public interface Conditions {
+
+	/**
+	 * Verifies that the consumed messages contain the expected message count.
+	 * @param messageCount the desired message count
+	 * @param <T> the type of the message
+	 * @return the condition
+	 */
+	static <T> Condition<T> desiredMessageCount(int messageCount) {
+		Assert.state(messageCount > 0, "Desired message count must be greater than 0");
+		return messages -> messages.size() == messageCount;
+	}
+
+	/**
+	 * Verifies that the consumed messages value contains at least the expected value.
+	 * @param expectation the expected value
+	 * @param <T> the type of the message
+	 * @return the condition
+	 */
+	static <T> Condition<T> containsValue(T expectation) {
+		return messages -> messages.stream().anyMatch(message -> message.getValue().equals(expectation));
+	}
+
+	/**
+	 * Verifies that the consumed messages value contains at least all expected values.
```

Comment:
This method confuses me a bit. Are we wanting to verify that all of the messages we received match the set of values. We expect messages "A", "B", "C" - did we get these? 

If so, I think we should rename this method to be more descriptive of what we are doing (same for the `containsValue` above). Wdyt? 

It may be helpful if we created an `AsssertableCondition` (like AssertJ) has so that users could just do something like:

```java
// Get 10 w/in 5 seconds or go boom
List<Message<UserPojo>> users = PulsarTestConsumer.consumeMessages(cf)
		.fromTopic("user-topic")
		.withSchema(Schema.JSON(UserPojo.class))
		.awaitAtMost(Duration.ofSeconds(5))
		.until((msgs) -> assertThat(msgs).containsExactlyInAnyOrder(...))
		.get();
```
as AssertJ is great (diverse API) for checking collection elements. 

I also think we could just wait to add this convenience factory until a 2nd iteration and simply leave it out for now. 

Thoughts?

Reply:
I agree that the methods might be better named. But I'm afraid that opening up `until` like this and not working with conditions and rather passing the in the received messages might confuse the end user even more. Should I assert or should I return something. While `ConsumedMessagesCondition<T>` presents a strong contract.